File: Symmetric_Tree/symmetric_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTreeModified(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if p == None and q == None:
            return True
        if not p or not q or p.val != q.val:
            return False
        return self.isSameTreeModified(p.right, q.left) and self.isSameTreeModified(p.left, q.right)
 
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return self.isSameTreeModified(root.left, root.right)
   

File: Symmetric_Tree/symmetric_tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return isSameTreeModified(root.right, root.left);
    }
    public boolean isSameTreeModified(TreeNode p, TreeNode q) {
        if (p == null && q == null) {
            return true;
        }
        if (q == null || p == null || p.val != q.val) {
            return false;
        }
        return isSameTreeModified(p.left, q.right) && isSameTreeModified(p.right, q.left);
    }
}
